server/policies/failing_policies.go

```diff
@@ -0,0 +1,215 @@
+// Package policies implements features to handle policy-related processing.
+package policies
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"net/url"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+// FailingPolicyAutomationType is the type of automations supported for
+// failing policies.
+type FailingPolicyAutomationType string
+
+// List of supported failing policy automation types.
+const (
+	FailingPolicyWebhook FailingPolicyAutomationType = "webhook"
+	FailingPolicyJira    FailingPolicyAutomationType = "jira"
+	FailingPolicyZendesk FailingPolicyAutomationType = "zendesk"
+)
+
+// FailingPolicyAutomationConfig holds the configuration for proessing a
+// failing policy to send to the configured automation.
+type FailingPolicyAutomationConfig struct {
+	AutomationType FailingPolicyAutomationType
+	PolicyIDs      map[uint]bool
+	WebhookURL     *url.URL // for webhook automation type only
+	HostBatchSize  int      // for webhook automation type only
+}
+
+// TriggerFailingPoliciesIntegration triggers an integration for failing
+// policies. It receives a function that takes care of sending the failed
+// policy as argument, that function receives the type of integration that is
+// enabled for that policy.
+func TriggerFailingPoliciesIntegration(
```

Comment:
This is the triggering logic for failing policies, independent of where the policy is sent (webhook, jira, etc.). It ensures that after being called, the processed redis failing policies sets are cleared appropriately on success.